packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -184,6 +184,61 @@ const setDifference = <T>(setA: Set<T>, setB: Set<T>): Set<T> => {
   return result;
 };
 
+/**
+ * Prepare python members for emission.
+ *
+ * If there are multiple members of the same name, they will all map to the same python
+ * name, so we will filter all deprecated members and expect that there will be only one
+ * left.
+ *
+ * Returns the members in a sorted list.
+ */
+function prepareMembers(members: PythonBase[], resolver: TypeResolver) {
+  // create a map from python name to list of members
+  const map: { [pythonName: string]: PythonBase[] } = {};
+  for (const m of members) {
+    let list = map[m.pythonName];
+    if (!list) {
+      list = map[m.pythonName] = [];
+    }
+
+    list.push(m);
+  }
+
+  for (const [name, list] of Object.entries(map)) {
+    // if the list of members for a name is 1, we are good
+    if (list.length === 1) {
+      continue;
+    }
+
+    // we found more than one member with the same python name, filter all
+    // deprecated versions and check that we are left with exactly one.
+    // otherwise, they will overwrite each other
+    // see https://github.com/aws/jsii/issues/2508
+    const nonDeprecated = list.filter((x) => !isDeprecated(x));
+    if (nonDeprecated.length > 1) {
+      throw new Error(
+        `Multiple non-deprecated members which map to the Python name "${name}"`,
+      );
+    }
+
+    // only retain the 1 non-deprecated member
+    map[name] = nonDeprecated;
+  }
+
+  // now return all the members
+  const ret = new Array<PythonBase>();
+  for (const v of Object.values(map)) {
+    if (v.length !== 1) {
+      throw new Error('assertion failed');
+    }
+
+    ret.push(v[0]);
+  }
```

Comment:
Seems like this whole loop can be part of the preceding loop, and offer a clearer error message.

Reply:
done.